### PR TITLE
Move connection metrics to ServiceDeployManager

### PR DESCRIPTION
**Pull request by reichlfl**
More pre-work for b/202262468 to reduce code duplication between the SessionSetup window and the new direct connection window.

### CODE REVIEW

File: src/SessionSetup/ServiceDeployManager.cpp [hunk 1]
```diff
@@ -560,9 +561,22 @@ void ServiceDeployManager::StartWatchdog() {
...
   outcome::result<GrpcPort> result = outcome::success(GrpcPort{0});
   DeferToBackgroundThreadAndWait(this, [&]() { result = ExecImpl(); });
+
+  if (result.error() == make_error_code(Error::kUserCanceledServiceDeployment)) {
+    connect_metric.SetStatusCode(orbit_metrics_uploader::OrbitLogEvent_StatusCode_CANCELLED);
+    LOG("OrbitService deployment has been aborted by the user");
+  } else {
+    connect_metric.SetStatusCode(orbit_metrics_uploader::OrbitLogEvent_StatusCode_INTERNAL_ERROR);
+    LOG("Deployment successful, grpc_port: %d", result.value().grpc_port);
+  }
```
Comment (antonrohr) [inline src/SessionSetup/ServiceDeployManager.cpp:578]: Here is something wrong. There are three possible outcomes of `ExecImpl` 
1. Success, in which case the `connect_metric` should either not be touched at all (the default status code is 
success), or should be set to SUCCESS explicitly.
2. The result carries an error, but the error code is kUserCanceledServiceDeployment. In this case : CANCELLED
3. The result is a different error, then its INTERNAL_ERROR

Also. Before `if (result.error())` there needs to be a `if (result.has_error())` or `if(!result)`

**Review by antonrohr — COMMENTED** [review]
In general looks good, but there is a problem with setting the status code and the if clause

File: src/SessionSetup/ServiceDeployManager.cpp (hunk 1 above)
Comment (reichlfl) [reply src/SessionSetup/ServiceDeployManager.cpp]: Thanks for catching this, I completely missed that!

**Review by antonrohr — APPROVED** [review]
LGTM!

**Pull request merged**